lib/molinillo/resolution.rb

```diff
@@ -88,8 +88,11 @@ def <=>(other)
         # @return [Integer] index of state requirement in reversed requirement tree
         #    (the conflicting requirement itself will be at position 0)
         def reversed_requirement_tree_index
-          return 999_999 unless state_requirement
-          requirement_tree.reverse.index(state_requirement)
+          @reversed_requirement_tree_index ||=
+            begin
+              return 999_999 unless state_requirement
```

Comment:
this will actually avoid the memoization.

```ruby
@reversed_requirement_tree_index = if state_requirement
  ....
else
  ...
```

should do the trick